syntax/internal/value/value.go

```diff
@@ -553,3 +555,15 @@ func convertGoNumber(nval Number, target reflect.Type) reflect.Value {
 
 	panic("unsupported number conversion")
 }
+
+func (v Value) Equal(rhs Value) bool {
+	if v.Type() != rhs.Type() {
+		return false
+	}
+
+	if !v.rv.Equal(rhs.rv) {
```

Comment:
Works for our shallow equal on strings, but may not work for maps/arrays perhaps where a DeepEqual would be needed? Maybe we should write a comment if that's the case.

Reply:
it panics for arrays : 
```
	t.Run("equal - array", func(t *testing.T) {
		v := value.Array(value.String("aa"), value.String("bb"))
		w := value.Array(value.String("aa"), value.String("bb"))
		require.True(t, v.Equal(w))
	})
```
panic: reflect.Value.Equal: values of type []value.Value are not comparable

Reply:
I documented the types that are not supported